### PR TITLE
Driver-only hashes: wrap-up

### LINKED ISSUE #7156: Update strategy.md for MD dispatch

### PR DESCRIPTION
**Pull request by mpg**
## Description

Resolve #7156

~**Depends on** #7327~ - merged, but the github cache hasn't been updated - the first new commit is "Update psa-migration/strategy.md"

## Gatekeeper checklist

- [x] **changelog** provided - for the entire series
- [x] **backport** not required - enhancement
- [x] **tests** not required - documentation-only PR

### CODE REVIEW

File: docs/architecture/psa-migration/strategy.md [hunk 1]
```diff
@@ -262,50 +272,32 @@ not possible to achieve good test coverage at the end of step 1 or step 2, it
...
-study, but in the meantime we can proceed with everything that's not the
-entropy module of one of the DRBG modules, and that does not depend on one of
-those modules.
+Regarding libmbedcrypto:
+- for hashes and ciphers, see `md-cipher-dispatch.md` in the same directory;
+- for ECC, we have no internal uses of the top-level algorithms (ECDSA, ECDH,
+  ECJPAKE), however they all depend on `ECP_C` which in turn depends on
+`BIGNUM_C`. So, direct calls from TLS, X.509 and PK to ECP and Bignum will
+need to be replaced; see <https://github.com/Mbed-TLS/mbedtls/issues/6839> and
+liked issues for a summary of intermediate steps and open points.
```
Comment (valeriosetti) [inline docs/architecture/psa-migration/strategy.md:300]: liked --> linked

**Review by valeriosetti — COMMENTED** [review]
LGTM, there is only 1 typo to fix.
Let's wait for preceding PRs to be merged then I'll approve it

**Review by valeriosetti — APPROVED** [review]

File: docs/architecture/psa-migration/psa-limitations.md [hunk 2]
```diff
@@ -11,11 +11,15 @@ is, of course, to actually do the migration work.
...
 
-Restartable ECC operations
---------------------------
+Restartable (aka interruptible) ECC operations
+----------------------------------------------
 
-There is currently no support for that in PSA at all, but it will be added at
-some point, see <https://github.com/orgs/Mbed-TLS/projects/1#column-18816849>.
+Support for interruptible ECDSA sign/verify was added to PSA in Mbed TLS 3.4.
+However support for interruptible ECDH is not present yet. Also, PK, X.509 and
```
Comment (mprse) [inline docs/architecture/psa-migration/psa-limitations.md:18]:
```suggestion
However, support for interruptible ECDH is not present yet. Also, PK, X.509 and
```

File: docs/use-psa-crypto.md [hunk 3]
```diff
@@ -11,16 +15,47 @@ General considerations
...
-code, which always uses PSA crypto. The parts of the TLS 1.3 code that will
-use PSA Crypto or not depending on the value of this option are:
+**Relationship with other options:** This option depends on
+`MBEDTLS_PSA_CRYPTO_C`. These two options differ in the following way:
+- `MBEDTLS_PSA_CRYPTO_C` enables the implementation of the PSA Crypto API.
+  When it is enabled, `psa_xxx()` APIs are available and you must call
+`psa_crypto_init()` before you call any other `psa_xxx()` function. Other
+modules in the library (non-PSA crypto APIs, X.509, TLS) may or may not use
+PSA Crypto but you're not required to call `psa_crypto_init()` before calling
+non-PSA functions, unless when explicitly documented (TLS 1.3).
```
Comment (mprse) [inline docs/use-psa-crypto.md:25]:
```suggestion
non-PSA functions, unless explicitly documented (TLS 1.3).
```

**Review by mprse — APPROVED** [review]
LGTM.

Just, very minor suggestions.

**Pull request merged**